File: 205-IsomorphicStrings/205-IsomorphicStrings.py
# Last updated: 3/31/2026, 9:39:04 PM
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        hS = [0]*200
        hT = [0]*200
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            if hS[ord(s[i])] != hT[ord(t[i])]:
                return False
            hS[ord(s[i])] = i+1
            hT[ord(t[i])] = i+1
        return True
        

File: 205-IsomorphicStrings/205-IsomorphicStrings.cpp
// Last updated: 3/31/2026, 9:38:58 PM
#define faster \
ios::sync_with_stdio(false);\
cout.tie(NULL);\
cin.tie(NULL);
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        faster;
        if (s.length() != t.length()){return false;};
        vector<int> hS(200, 0);
        vector<int> hT(200, 0);
        for(int i=0;i<s.length();i++){
            if (hS[s[i]] != hT[t[i]]){
                return false;
            }
            hS[s[i]] = i+1;
            hT[t[i]] = i+1;
        }
        return true;
    }
};